server/services/timelineService.ts

```diff
@@ -1,28 +1,83 @@
-import type { Timeline } from 'viewModels'
-import type { TimelineResponse } from 'educationAndWorkPlanApiClient'
+import type { Prison, Timeline, TimelineEvent } from 'viewModels'
+import type { TimelineEventResponse } from 'educationAndWorkPlanApiClient'
 import EducationAndWorkPlanClient from '../data/educationAndWorkPlanClient'
 import { toTimeline } from '../data/mappers/timelineMapper'
 import logger from '../../logger'
+import PrisonService from './prisonService'
+import { HmppsAuthClient } from '../data'
+
+const SUPPORTED_TIMELINE_EVENTS = ['ACTION_PLAN_CREATED', 'INDUCTION_UPDATED']
 
 export default class TimelineService {
-  constructor(private readonly educationAndWorkPlanClient: EducationAndWorkPlanClient) {}
+  constructor(
+    private readonly hmppsAuthClient: HmppsAuthClient,
+    private readonly educationAndWorkPlanClient: EducationAndWorkPlanClient,
+    private readonly prisonService: PrisonService,
+  ) {}
 
-  async getTimeline(prisonNumber: string, token: string): Promise<Timeline> {
+  async getTimeline(prisonNumber: string, token: string, username: string): Promise<Timeline> {
     try {
+      const systemToken = await this.hmppsAuthClient.getSystemClientToken(username)
```

Comment:
OK, this is why we need the username - we use it in order to get a system token. We need the system token to pass into the `PrisonService` so that it can retrieve prisons from the `prison-register-api` if it needs to